src/modules/article-favorites/article-favorites.service.ts

```diff
@@ -0,0 +1,27 @@
+import { Injectable } from '@nestjs/common';
+import {ArticleFavoriteRepository} from "@module/article-favorites/repositories/article-favorites.repository";
+
+@Injectable()
+export class ArticleFavoritesService {
+  constructor(private readonly articleFavoriteRepository: ArticleFavoriteRepository) {}
+
+  async favoriteArticle(userId: number, articleId: number) {
+    const alreadyFav = await this.articleFavoriteRepository.isFavorited(userId, articleId);
+
+    if (!alreadyFav) {
```

Comment:
- Nên move phần xử lý kiểm tra điều kiện thuận lên trc ạ: bổ sung response trả về message khi favorite thành công
```suggestion
    if (alreadyFav) {
```
- I18n nội dung message